src/main/java/core/basesyntax/service/operation/ReturnOperation.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.dao.FruitTransactionDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class ReturnOperation implements OperationHandler {
+    private final FruitTransactionDao fruitTransactionDao;
+
+    public ReturnOperation(FruitTransactionDao fruitTransactionDao) {
+        this.fruitTransactionDao = fruitTransactionDao;
+    }
+    @Override
+    public void recount(FruitTransaction transaction) {
+        if (transaction.getQuantity() < 0) {
```

Comment:
It's good that you're checking for negative quantities, but consider moving this validation logic to a separate method to make the `recount` method focused on a single responsibility, adhering to the SOLID principles.